lua/lspsaga/diagnostic.lua

```diff
@@ -59,7 +59,7 @@ local show_diagnostics = function(opts, get_diagnostics)
   end
 
   for i, diagnostic in ipairs(diagnostics) do
-    local prefix = string.format("%d. ", i)
+    local prefix = config.prefix_diagnostic and string.format("%d. ", i) or ""  
```

Comment:
Move part of the condition outside the loop
```lua
-- line 43
local diagnostic_prefix = config.prefix_diagnostic and "%d. " or ""
```
then in place, more readable and faster 
```lua
local prefix = string.format(diagnostic_prefix, i)
```

Reply:
I've added config.diagnostic_prefix_format="%d. "  instead, but I've had to recreate my repo. Will close this PR and raise another.